### PR TITLE
AYR-1238 - Fixes for OpenSearch sorting query

### PR DESCRIPTION
**Pull request by ltrasca**
## Changes in this PR

- Modified existing function that built the OpenSearch sort query to map our search terms to value names (e.g. closure_type is mapped to metadata.closure_type.keyword) - this has a secondary function of allowing us to check that a sort term should be added to the query or not, random terms should not be added to it
- Fixed dropdown component not being passed a correct object with sort terms & subsequently not showing the currently selected term
- Added unit tests to handle some edge cases and check for dropdown values when a sort term is selected

## JIRA ticket
https://national-archives.atlassian.net/browse/AYR-1238

## Screenshots of UI changes

N/A

- [ ] Requires env variable(s) to be updated

### CODE REVIEW

File: app/main/routes.py [hunk 1]
```diff
@@ -587,7 +587,9 @@ def search_transferring_body(_id: uuid.UUID):
     page = int(request.args.get("page", 1))
 
     filters = {"query": query}
-    sorting_orders = build_sorting_orders_open_search(request.args)
+
+    sorting_query = build_sorting_orders_open_search(request.args)
+    sorting_orders = build_sorting_orders(request.args)
```
Comment (anthonyhashemi) [inline app/main/routes.py:592]: `sorting_orders` isnt being used in this function anymore so I think this line can be removed

File: app/tests/test_search.py [hunk 2]
```diff
@@ -1406,6 +1483,15 @@ def test_search_transferring_body_with_search_filter_full_test(
...
         assert response.status_code == 200
         soup = BeautifulSoup(response.data, "html.parser")
+        select = soup.find("select", {"id": "sort"})
+        option = select.find("option", selected=True)
+        option_text = "series_id-asc"
+
+        if option:
+            option_text = option.get("value")
+
+        assert select
```
Comment (anthonyhashemi) [inline app/tests/test_search.py:1537]: what are we asserting here?

File: app/tests/test_search.py [hunk 3]
```diff
@@ -1406,6 +1483,15 @@ def test_search_transferring_body_with_search_filter_full_test(
 
         assert response.status_code == 200
         soup = BeautifulSoup(response.data, "html.parser")
+        select = soup.find("select", {"id": "sort"})
+        option = select.find("option", selected=True)
+        option_text = "series_id-asc"
+
+        if option:
```
Comment (anthonyhashemi) [inline app/tests/test_search.py]: Can we break this out into 2 tests, one where option is expected and another where it is not, so that each test is testing a distinct scenario so it is easier to debug and reason with

File: app/tests/test_search.py (hunk 2 above)
Comment (ltrasca) [reply app/tests/test_search.py:1537]: just that the select exists

File: app/tests/test_search.py (hunk 3 above)
Comment (anthonyhashemi) [reply app/tests/test_search.py]: I saw the updated commit but looks like we still have that condition

Comment (ltrasca) [reply app/tests/test_search.py]: @anthonyhashemi pushed a commit with the changes

**Review by anthonyhashemi — APPROVED** [review]
